### PR TITLE
Prevent multiple membership cancellation emails

### PR DESCRIPTION
**Pull request by timEulitz**
https://phabricator.wikimedia.org/T226958

### CODE REVIEW

File: src/UseCases/CancelMembershipApplication/CancelMembershipApplicationUseCase.php [hunk 1]
```diff
@@ -40,16 +40,16 @@ public function cancelApplication( CancellationRequest $request ): CancellationR
...
+			}
+			$this->sendConfirmationEmail( $application );
 		}
-		catch ( StoreMembershipApplicationException $ex ) {
-			return $this->newFailureResponse( $request );
-		}
-
-		$this->sendConfirmationEmail( $application );
 
 		return $this->newSuccessResponse( $request );
```
Comment (tzhelyazkova) [inline src/UseCases/CancelMembershipApplication/CancelMembershipApplicationUseCase.php]: Not feeling too good about returning success response in either case but .. I guess it's not that big of a deal.

Comment (gbirke) [reply src/UseCases/CancelMembershipApplication/CancelMembershipApplicationUseCase.php]: From a usability perspective I think it's better to be able to click the "cancel" button multiple times without anything bad happening or having a condescending "you already canceled" message.

Comment (gbirke) [reply src/UseCases/CancelMembershipApplication/CancelMembershipApplicationUseCase.php]: On a more conceptual level - seeing the domain object methods as state transitions and making sure the domain objects allow only valid transitions and states, I'd have opted for making the `Application::cancel` method to check if it's already canceled and throw a DomainException if something cancels twice. 

When this code was written I did not have a clear concept of those DDD concepts and the entities are more "dumb" and the logic is in the use cases. We don't need to change that in this small PR, just keep it in mind for the future.

**Comment by gbirke** [comment]
This PR should have a follow-up with unit tests: 

    Given a canceled application, the use case
    * does not cancel the application
    * does not stores the application
    * does not send an email
    * returns a success response

**Pull request merged**